symengine/codegen.h

```diff
@@ -232,6 +232,8 @@ class C89CodePrinter : public BaseVisitor<C89CodePrinter, CodePrinter>
     {
         if (eq(*b, *rational(1, 2))) {
             o << "sqrt(" << apply(a) << ")";
+        } else if (eq(*a, *E)) {
```

Comment:
Shouldn't this come before? `exp(1/2)` is better than `sqrt(E)`

Reply:
@bjodah, what do you think?

Reply:
I don't think it matters (even at -O0):
```
#!/bin/bash                                                                                                     
function get_assembly() {
        gcc $CFLAGS -S -x c -o - - <<EOF                                                                        
#include <math.h>                                                                                               
double f(double x) { return x*$1; }                                                                             
EOF                                                                                                             
}
result_a=$(get_assembly "sqrt(M_E)")
result_b=$(get_assembly "exp(0.5)")
diff <(echo $result_a) <(echo $result_b)
```
running:
```
$ CFLAGS=-O0 ./compare_assembly.sh
```
gives no diff

Reply:
So you're right. `exp(0.5)` is preferable since `M_E` isn't in the ISO standard (and on msvc math macros need to be opt-ed in using a macro). One possible "pragmatic" way to deal with this is to limit the use of math macros to the C99 printer and avoid them in the C89 printer.